driver-core/src/test/java/com/datastax/driver/core/policies/CASRetryPolicyIntegrationTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.policies;
+
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.exceptions.UnavailableException;
+import com.datastax.driver.core.utils.CassandraVersion;
+import org.testng.annotations.Test;
+
+import static com.datastax.driver.core.Assertions.assertThat;
+import static com.datastax.driver.core.ConsistencyLevel.ANY;
+import static com.datastax.driver.core.ConsistencyLevel.SERIAL;
+import static com.datastax.driver.core.CreateCCM.TestMode.PER_METHOD;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+
+/**
+ * Tests for retry policies in combination with lightweight transactions.
+ * <p/>
+ * Note: we can't extend {@link AbstractRetryPolicyIntegrationTest} here, because we are
+ * testing LWT statements and Scassandra doesn't support priming consistency levels yet.
```

Comment:
I think the's tests need to be changed now that we retry once on unavailable, and then there are no remaining hosts to try, so we get an NHAE